annotations/knative-annotations/src/test/java/io/dekorate/knative/config/KnativeConfigTest.java

```diff
@@ -25,9 +25,7 @@ public void testGeneratedCode() {
         .withName("myapp")
         .withVersion("1.0.0")
         .withExpose(true)
-        .addNewLabel("key1", "val1")
+        .addNewLabel("key1", "val1", new String[0])
```

Comment:
Do we really need to do this? Can't the existing `addNewLabel` continue to exist?

Reply:
No, it can't.

Possible alternatives:

```java
.addNewLabel().withKey("key1").withValue("val1").endLabel()
```

Given that this change is not user facing, I'd say its no biggie.

Reply:
Sure yeah, it just looks ugly

Reply:
autoformatting doesn't work as expected.

Reply:
@geoand: The main thing with these methods is that they are generated. 

The logic is the following:
`if the object is flat and has less than five 5 fields generate a with method with all fields`.

In theory we could create a method for each possible field combination. In the current case it would look like:

```
withNewLabel(String key);
withNewLabel(String key, String value); 
withNewlabel(String key, String value, String[] kinds);
```
However, this would mean that we would also get methods that don't make sense. 
Now, since to original object is an annotation and fields without default values are considered mandatory we could possibly work soemthing out.

I think that this would be one or two days work. and I'd like to avoid it at the moment.
Created: https://github.com/sundrio/sundrio/issues/177

Reply:
@geoand if this sounds reasonable can we approve this?